go-controller/pkg/ovn/endpoints.go

```diff
@@ -93,7 +93,26 @@ func (ovn *Controller) AddEndpoints(ep *kapi.Endpoints) error {
 			}
 			vip := util.JoinHostPortInt32(svc.Spec.ClusterIP, svcPort.Port)
 			ovn.AddServiceVIPToName(vip, svcPort.Protocol, svc.Namespace, svc.Name)
-			ovn.handleExternalIPs(svc, svcPort, lbEps.IPs, lbEps.Port, false)
+			if len(svc.Spec.ExternalIPs) > 0 {
+				gateways, _, err := ovn.getOvnGateways()
+				if err != nil {
+					return err
+				}
+				for _, gateway := range gateways {
+					loadBalancer, err := ovn.getGatewayLoadBalancer(gateway, svcPort.Protocol)
+					if err != nil {
+						klog.Errorf("physical gateway %s does not have load_balancer "+
+							"(%v)", gateway, err)
+						continue
+					}
+					if loadBalancer == "" {
+						continue
```

Comment:
shouldn't there be an error logged here if we get back no loadbalancer? Is this supposed to be the error on line 104?

Reply:
I didn't remove the check. I just placed it in `getGatewayLoadBalancer` so that we skip repeating ourselves all over the code base for a condition we obviously want to avoid.

Reply:
@alexanderConstantinescu the check at L104-105 can be removed with your changes. If the returned loadBalancer is indeed empty, then you return `err', so the code will hit L100 and at L104 it will never be the case that loadBalancer will be empty.